web_dashboard_dafa/models/af_dashboard.py

```diff
@@ -79,11 +126,26 @@ class Dashboard(models.Model):
     name = fields.Char("Name")
     user_id = fields.Many2one('res.users', "User")
     firstname = fields.Char(related='user_id.firstname', store=True)
-    welcome_msg = fields.Char("Welcome String", default="Welcome")
+    welcome_msg = fields.Char("Welcome String", default="Välkommen")
     welcome_dsc = fields.Html("Welcome Description")
     box_ids = fields.One2many('dashboard.boxes', 'af_dashboard_id')
     group_ids = fields.Many2many('res.groups', related='user_id.groups_id')
 
+    def update_welcome_string(self):
+        for dashboard in self:
+            dashboard.welcome_msg = "Välkommen"
+
+    def check_duplicate_boxes(self):
+        for dashboard in self:
+            for box in dashboard.box_ids:
+                if box.description and box.user_id and not box.hide:
+                    cleanr = re.compile('<.*?>')
+                    same_boxes = dashboard.box_ids.filtered(lambda b: b.id != box.id)
```

Comment:
Shouldn't this be b.id == box.id ? (i'm not sure just double check this)

Reply:
It is correct. I have checked.